index.js

```diff
@@ -28,15 +28,21 @@ exports.register = function ( server, options, next ) {
         request.i18n.setLocale(languageCode);
       }
     }
+
+    if (pluginOptions.queryParameter && request.query && request.query[pluginOptions.queryParameter]) {
+        request.i18n.setLocale(request.query[pluginOptions.queryParameter]);
+    }
+
```

Comment:
The query parametter is set at the plugin option and the path parameter is set at the route configuration.
It seemed to me to manage the priority according to where the choice of the language is settled